### DISCUSSION TITLE
Related jwlibrary merge projects

### DISCUSSION THREAD

**Discussion by abdon-copper** [AndreasSko/go-jwlm#48] [General]
Hello Andreas,

## Review

Today, I tried your JWLM app on an iPad. The tutorial is very clear and the GUI looks good and polished. Great conflict resolution options. It's fast and merges files locally.

Feedback:

1. The merge button is still visible after the merge is finished, and it wasn't intuitive that I had to click the export button next.
2. The workflow to merge more than 2 files is certainly possible by exporting to local file system and adding that to import files again, however you could consider to build in support to merge 3 files at once.
3. It's not that clear how a user should start a new merge, apparently you can click the large rectangle with counts again to import a different file.
4. I would also suggest to set the manifest name to a more unique identifier instead of the static string 'go-jwlm', maybe include the original device names that were merged.
5. As you already noticed, OneDrive doesn't work for me either

In conclusion, I can certainly recommend your app to iPhone and iPad owners that want to sync their backup files.

## Related projects

I'm the author of a similar project that targets WebAssembly instead of iOS. A third project exists, developed for the Windows desktop. We could help each other by sharing the database schema documentation and good conflict merge strategies. So I just wanted to let you these repositories exist if you didn't find them already on Github.

All these 3 projects support database schema v8 and can merge notes, bookmarks and markings. Each of them has both a command-line interface and a graphical user interface.

[merge-jwl]: https://merge-jwl.netlify.app/
[JWLM]: https://apps.apple.com/ie/app/jwlm-jw-library-merger/id1539780103
[JWLMerge]: https://github.com/AntonyCorbett/JWLMerge/releases

| App | [JWLMerge] | [merge-jwl] | [JWLM] |
|-|-|-|-|
| Source | [AntonyCorbett/JWLMerge](https://github.com/AntonyCorbett/JWLMerge) | [abdon-gadgets/merge-jwl](https://github.com/abdon-gadgets/merge-jwl) | [AndreasSko/ios-jwlm](https://github.com/AndreasSko/ios-jwlm) |
| Author | Antony Corbett | Abdon | Andreas Skorczyk |
| First release | Jan 2018 | Sep 2020 | Nov 2020 |
| Platforms | Windows, Wine | Web Browser | iOS |
| Programming languages | C# | Rust, TypeScript | Go, Swift |
| Translations | English | English | English, German |

Didn't test yet:

1. https://github.com/Astasian/jwlmerge-pwa
2. https://gitlab.com/erykj/jwlmanager

**Comment by AntonyCorbett** [comment]
Chart correction - JWLMerge source is AntonyCorbett/JWLMerge
